bot/cogs/duck_pond.py

```diff
@@ -0,0 +1,174 @@
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.webhook_id = constants.Webhooks.duck_pond
+        self.bot.loop.create_task(self.fetch_webhook())
+
+    async def fetch_webhook(self) -> None:
+        """Fetches the webhook object, so we can post to it."""
+        await self.bot.wait_until_ready()
+
+        try:
+            self.webhook = await self.bot.fetch_webhook(self.webhook_id)
+        except discord.HTTPException:
+            log.exception(f"Failed to fetch webhook with id `{self.webhook_id}`")
+
+    @staticmethod
+    def is_staff(member: Union[User, Member]) -> bool:
+        """Check if a specific member or user is staff."""
+        if hasattr(member, "roles"):
+            for role in member.roles:
+                if role.id in constants.STAFF_ROLES:
+                    return True
+        return False
+
+    async def has_green_checkmark(self, message: Message) -> bool:
+        """Check if the message has a green checkmark reaction."""
+        for reaction in message.reactions:
+            if reaction.emoji == "✅":
+                async for user in reaction.users():
+                    if user == self.bot.user:
+                        return True
+        return False
+
+    async def send_webhook(
+        self,
+        content: Optional[str] = None,
+        username: Optional[str] = None,
+        avatar_url: Optional[str] = None,
+        embed: Optional[Embed] = None,
+    ) -> None:
+        """Send a webhook to the duck_pond channel."""
+        try:
+            await self.webhook.send(
+                content=content,
+                username=username,
+                avatar_url=avatar_url,
+                embed=embed
+            )
+        except discord.HTTPException:
+            log.exception("Failed to send a message to the Duck Pool webhook")
+
+    async def count_ducks(self, message: Message) -> int:
+        """
+        Count the number of ducks in the reactions of a specific message.
+
+        Only counts ducks added by staff members.
+        """
+        duck_count = 0
+        duck_reactors = []
+
+        for reaction in message.reactions:
+            async for user in reaction.users():
+
+                # Is the user a staff member and not already counted as reactor?
+                if not self.is_staff(user) or user.id in duck_reactors:
+                    continue
+
+                # Is the emoji a duck?
+                if hasattr(reaction.emoji, "id"):
+                    if reaction.emoji.id in constants.DuckPond.custom_emojis:
+                        duck_count += 1
+                        duck_reactors.append(user.id)
+                elif isinstance(reaction.emoji, str):
+                    if reaction.emoji == "🦆":
+                        duck_count += 1
+                        duck_reactors.append(user.id)
+        return duck_count
+
+    async def relay_message_to_duck_pond(self, message: Message) -> None:
```

Comment:
This name feels too verbose. How about `relay_messages`?

Reply:
That's okay with me. I liked the verbosity when reading the name, as it spells out what's going to happen at that point in the code and `relay_message` feels more generic (which this method is not). It doesn't really matter, though, so I'll change it.